src/EFCore.PG/Infrastructure/Internal/NpgsqlOptionsExtension.cs

```diff
@@ -118,6 +127,19 @@ public override bool ApplyServices(IServiceCollection services)
         // See https://github.com/aspnet/EntityFrameworkCore/pull/10091
         public override int? MinBatchSize => base.MinBatchSize ?? 2;
 
+        /// <summary>
+        /// Returns a copy of the current instance configured with the specified range mapping.
+        /// </summary>
+        [NotNull]
+        internal virtual NpgsqlOptionsExtension WithRangeMapping(string rangeName, Type elementClrType, string subtypeName)
+        {
+            var clone = (NpgsqlOptionsExtension)Clone();
+
+            clone._rangeMappings.Add((rangeName, elementClrType, subtypeName));
+
+            return clone;
+        }
```

Comment:
nit: Does these empty lines improve readability?

Reply:
Not really, but this is the way it was when copied from EF Core... Let's leave the same for now (we can clean up later)